docs: Update and remove deprecated documentation, including new publi…

JackHowa opened the pull request:
- Emphasize rebasing after merging into new branch and publishing https://github.com/WPMedia/fusion-news-theme-blocks/pull/596
- Remove outdated docs
- Referred to steps docs in fusion news theme step-by-step guide

Commit:
docs: Update and remove deprecated documentation, including new publish workflow

beltrancaliz commented on News Theme Development.md:
```diff
@@ -54,14 +42,12 @@ The news-theme-css repo is located at: <https://github.com/WPMedia/news-theme-cs
 
 news-theme-css is provided to the other parts of the system as an NPM package. For more information on the build and publishing process for this repository, please see the readme.md for the project: <https://github.com/WPMedia/news-theme-css/blob/stable/readme.md>
 
-**Note:** When publishing, you will need a .npmrc file that gives you access to the private NPM repo. Reach out to a team member to get this.
```
Maybe we should reword this just to make sure that clients have it when they create a bundle (even if we don't go through the steps for creating the token here)

JackHowa replied:
good idea

JackHowa replied:
updated

beltrancaliz commented on News Theme Development.md:
```diff
@@ -168,26 +129,15 @@ git checkout -b PEN-[jira ticket num]-[brief description of feature]
-Before publishing, make sure a Pull Request has been made and merged from the `canary` branch against the `beta` branch. This PR should only include features that belong to the current release, so make sure to merge as soon as the features for the release have been completed to avoid including further features from `canary` belonging to a later release.
-
----
+Merge into `canary` branch to publish to canary tag. Please reach out to arc block maintainers to talk about publishing into `beta` or `stable` or other tags.
```
add `rc`?

JackHowa replied:
yep updated

beltrancaliz reviewed (changes requested):
Added a few suggestions

Commit:
docs: Update per comment to clarify the use of the npmrc

No longer need it for the private npm registry via npm

https://github.com/WPMedia/fusion-news-theme-blocks/pull/597#discussion_r534989950

Commit:
docs: Clarify docs tag and show where to find more info

JackHowa commented:
@beltrancaliz updated -- thanks for review
